City-of-Helsinki/hitas
# Maximum price calculation additional info field

# Hitas Pull Request

# Description

- Add `additional_field` to confirmed and unconfirmed maximum price PDFs.
- For confirmed maximum price, it's entered when creating a calculation.
- For unconfirmed (estimate) it's entered before downloading the PDF.
  - The old button now opens a modal, with 'Lisätieto'-field. The value is sent to the API and the text is inserted in the PDF when generating it.

## Pull request checklist

Check the boxes for each DoD items that has been completed:

- **Frontend**
- **Backend**
    - [x] Automatic tests has been added
    - [x] Database migrations will work in test environment

## Test plan

- Create a confirmed price calculation and make sure that the json on the pdf contains `additional_info` and the text you entered
- Download an unconfirmed price PDF, and enter some text in the model, make sure the PDF contains the text you entered.

## Tickets

This pull request resolves all or part of the following ticket(s): HT-329
